components/forms/radio/index.jsx

```diff
@@ -79,7 +83,8 @@ class Radio extends React.Component {
 					id={this.getId()}
 					name={this.props.name}
 					value={this.props.value}
-					checked={!this.props.disabled && this.props.checked}
```

Comment:
@interactivellama if we kept this logic this component would always be controlled no mater what, `disabled` should not affect the checked state.